test/env/lib.rs

```diff
@@ -326,3 +473,76 @@ pub mod icrc1 {
         ledger.query("icrc1_fee", ()).await.map(|(t,)| t)
     }
 }
+
+pub fn standard_replica_burn_fn(
+    agent: Arc<Agent>,
+    canister_id: Principal,
+    amount: Nat,
+) -> BurnReturnType {
+    // This method panics if it is called but burning is not supported by the given icrc1 ledger
+    Box::pin(async move {
+        let minting_account = Decode!(
+            agent
+                .query(&canister_id, "icrc1_minting_account")
+                .with_arg(Encode!(&()).unwrap())
+                .call()
+                .await?
+                .as_slice(),
+            Option<Account>
+        )
+        .unwrap()
+        .unwrap();
+        let res =
+            Decode!(agent.update(&canister_id, "icrc1_transfer").with_arg(Encode!(&Transfer::amount_to(amount,minting_account.owner)
```

Comment:
Why you are passing just the owner?  You need to pass the full `minting_account` including the subaccount otherwise Ledgers that use Accounts with no default subaccount will get the wrong burn function, no?